runtime/compiler/x/codegen/J9CodeGenerator.cpp

```diff
@@ -85,19 +85,23 @@ J9::X86::CodeGenerator::CodeGenerator() :
    cg->setSupportsPartialInlineOfMethodHooks();
    cg->setSupportsInliningOfTypeCoersionMethods();
    cg->setSupportsNewInstanceImplOpt();
-   if (cg->getX86ProcessorInfo().supportsSSE4_1() &&
+   
+   TR_ASSERT(cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_SSE4_1) == cg->getX86ProcessorInfo().supportsSSE4_1(), "supportsSSE4_1() failed\n");
```

Comment:
These asserts will go away as far I understand, so we could replace them with TR_ASSERT_FATAL without worrying about extra overhead forever.